src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkValueProvider.cs

```diff
@@ -207,17 +212,27 @@ private async Task<bool> IsWPFApplicationAsync(IProjectProperties defaultPropert
                 {
                     // Enabled
 
-                    // Create the Application.xaml if it doesn't exist. We don't care about the path, we just need it to be created.
-                    string? appXamlFilePath = await GetAppXamlRelativeFilePathAsync(create: true);
-                    if (appXamlFilePath is not null)
+                    //project accessor to remove the applicationdefinition property if re-enabled
+                    await _projectAccessor.OpenProjectXmlForWriteAsync(_project, project =>
                     {
-                        IEnumerable<IProjectItem> matchingItems = await _sourceItemsProvider.GetItemsAsync(NoneItemType, appXamlFilePath);
-                        IProjectItem? appXamlItem = matchingItems.FirstOrDefault();
-                        if (appXamlItem is not null)
+                        foreach (ProjectItemGroupElement itemGroup in project.ItemGroups.ToList())
```

Comment:
Is the `ToList` necessary here? Unlike for `Items`, we aren't changing the `ItemGroups` property in the loop, so I don't think we need a defensive copy to be made before enumerating.